classes/i18n/Locale.php

```diff
@@ -513,4 +556,29 @@ private function _getSupportedLocales(): array
             ?? array_map(fn (LocaleMetadata $locale) => $locale->locale, $this->getLocales());
         return $this->supportedLocales = array_combine($locales, $locales);
     }
+
+    /**
+     * Get Weblate submission languages to array
+     * Combine app's language names with weblate's in English.
+     * Weblate's names override app's if same locale key
+     * 
+     * @return string[]
+     */
+    private function _getSubmissionLocaleNames(): array
+    {
+        return $this->submissionLocaleNames ??= (function (): array {
+            $file = Core::getBaseDir() . '/' . PKP_LIB_PATH . '/lib/weblateLanguages/languages.json';
+            $key = __METHOD__ . self::MAX_CACHE_LIFETIME . sha1($file);
```

Comment:
I think using sha1 of the file path will not change the cache i. e. create a new cache if the content of the file has changed, right? If so, msybe to use filemtime instead?

Reply:
We can maybe introduce a new CACHE_LIFETIME period = 1year or so to use here - - I think the cache dors not need to change so often, only when the content of that file with languages changrs, but it should also not last forever...